owasp-top10-2017-apps/a10/games-irados/app/routes.py

```diff
@@ -60,29 +67,37 @@ def decorated_function(*args, **kwargs):
         return f(*args, **kwargs)
     return decorated_function
 
+
 @app.route('/', methods=['GET'])
 def root():
     return redirect('/login')
 
+
 @app.route('/logout', methods=['GET'])
 @login_required
 def logout():
     session.clear()
     return redirect('/login')
 
+
 @app.route('/login', methods=['GET', 'POST'])
 def login():
     if request.method == 'POST':
+        if not is_human(request):
+            flash("Captcha inválido!.", "danger")
+            return render_template('login.html', sitekey=RECAPTCHA_SITE_KEY)
         username = request.form.get('username').encode('utf-8')
         psw = Password(request.form.get('password').encode('utf-8'))
         user_password, success = database.get_user_password(username)
         if not success or user_password == None or not psw.validate_password(str(user_password[0])):
             flash("Usuario ou senha incorretos", "danger")
-            return render_template('login.html')
+            app.logger.warning('[IP:%s][User:%s][Method:%s][Path:%s] Failed to login', request.remote_addr, username.decode(), request.method , request.path)
```

Comment:
When an user tries to login mistakenly using his password as his username, your log will register this sensitive data: `[2019-02-26 20:07:34,449] WARNING in routes: [IP:192.168.32.1][User:password123][Method:POST][Path:/login] Failed to login`. 

What do you think? Would you mind removing it from your code? 😃

Reply:
Fixed!  =D

Reply:
Very creative fix! 👏🏻 You rock! 🔥